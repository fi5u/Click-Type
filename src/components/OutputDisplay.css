.OutputDisplay {
    border: 2px solid #999;
    border-radius: 4px;
    margin: 0 auto;
    min-width: 50%;
    min-height: 120px;
    padding: 10px;
    text-align: left;
}

.OutputDisplay__content {
    position: relative;
    white-space: pre-wrap;
}

.OutputDisplay__content:after {
    animation: flashcursor .9s infinite;
    background-color: #666;
    content: "";
    position: absolute;
    height: 20px;
    width: 1px;
}

@keyframes flashcursor {
    0% { opacity: 0; }
    49.9% { opacity: 0; }
    50% { opacity: 1; }
    99.9% { opacity: 1; }
}
