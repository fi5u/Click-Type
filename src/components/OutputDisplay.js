import './OutputDisplay.css'
import React, {
    Component,
    PropTypes,
} from 'react'

class OutputDisplay extends Component {
    render() {
        const {
            value
        } = this.props

        return(
            <div
                className="OutputDisplay"
            >
                <span
                    className="OutputDisplay__content"
                >
                    {value}
                </span>
            </div>
        )
    }
}

OutputDisplay.propTypes = {
    value: PropTypes.string.isRequired,
}

export default OutputDisplay
