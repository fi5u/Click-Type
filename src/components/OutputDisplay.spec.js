/* global expect, it, jest */
import OutputDisplay from './OutputDisplay'
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

const props = {
    value: '',
}

it('renders OutputDisplay without crashing', () => {
    shallow(
        <OutputDisplay {...props} />
    )
})

it('renders OutputDisplay correctly', () => {
    let component = renderer.create(
        <OutputDisplay {...props} />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    // Add text
    component = renderer.create(
        <OutputDisplay
            {...props}
            value="The quick brown fox"
        />
    )
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
