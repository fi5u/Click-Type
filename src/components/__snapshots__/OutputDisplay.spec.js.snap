exports[`test renders OutputDisplay correctly 1`] = `
<div
  className="OutputDisplay">
  <span
    className="OutputDisplay__content">
    
  </span>
</div>
`;

exports[`test renders OutputDisplay correctly 2`] = `
<div
  className="OutputDisplay">
  <span
    className="OutputDisplay__content">
    The quick brown fox
  </span>
</div>
`;
